addons/project/wizard/project_share_wizard.py

```diff
@@ -43,13 +43,30 @@ def _compute_resource_ref(self):
             else:
                 wizard.resource_ref = None
 
-    def action_send_mail(self):
+    def action_share_record(self):
+        # Confirmation dialog is only opened if new portal user(s) need to be created in a 'on invitation' website
         self.ensure_one()
+        on_invite = self.env['res.users']._get_signup_invitation_scope() == 'b2b'
+        new_portal_user = on_invite and any(self.partner_ids.filtered(lambda p: not p.user_ids))
```

Comment:
What's the purpose of the `any` here?
If you want lazy evaluation, it won't work with filtered, but I believe you can do something like this:
```suggestion
        new_portal_user = on_invite and next((not p.user_ids for p in self.partner_ids), False)
```

Reply:
> What's the purpose of the `any` here? If you want lazy evaluation, it won't work with filtered, but I believe you can do something like this:
Purpose of `any` was only to get a boolean, I guess I've could just use `bool` as well.